### PR TITLE
Style tag support.

### PR DESCRIPTION
**Pull request by cwelch5**

### CODE REVIEW

File: src/Helmet.js [hunk 1]
```diff
@@ -276,9 +279,9 @@ const generateTagsAsString = (type, tags) => {
             })
             .join(" ").trim();
 
-        const innerHTML = tag.innerHTML || "";
+        const tagContent = tag.innerHTML || tag.cssText || "";
 
-        return `<${type} ${HELMET_ATTRIBUTE}="true" ${attributeHtml}${type === TAG_NAMES.SCRIPT ? `>${innerHTML}</${type}>` : `/>`}`;
+        return `<${type} ${HELMET_ATTRIBUTE}="true" ${attributeHtml}${type === TAG_NAMES.SCRIPT ? `>${tagContent}</${type}>` : `/>`}`;
```
Comment (doctyper) [inline src/Helmet.js]: This is only checking for `TAG_NAMES.SCRIPT`, do you need to add `TAG_NAMES.STYLE`?

File: src/Helmet.js [hunk 2]
```diff
@@ -423,15 +430,25 @@ const Helmet = (Component) => {
 const reducePropsToState = (propsList) => ({
     htmlAttributes: getHtmlAttributesFromPropsList(propsList),
     title: getTitleFromPropsList(propsList),
-    onChangeClientState: getOnChangeClientState(propsList),
     baseTag: getBaseTagFromPropsList([TAG_PROPERTIES.HREF], propsList),
     metaTags: getTagsFromPropsList(TAG_NAMES.META, [TAG_PROPERTIES.NAME, TAG_PROPERTIES.CHARSET, TAG_PROPERTIES.HTTPEQUIV, TAG_PROPERTIES.PROPERTY], propsList),
     linkTags: getTagsFromPropsList(TAG_NAMES.LINK, [TAG_PROPERTIES.REL, TAG_PROPERTIES.HREF], propsList),
-    scriptTags: getTagsFromPropsList(TAG_NAMES.SCRIPT, [TAG_PROPERTIES.SRC, TAG_PROPERTIES.INNER_HTML], propsList)
+    scriptTags: getTagsFromPropsList(TAG_NAMES.SCRIPT, [TAG_PROPERTIES.SRC, TAG_PROPERTIES.INNER_HTML], propsList),
+    styleTags: getTagsFromPropsList(TAG_NAMES.STYLE, [TAG_PROPERTIES.CSS_TEXT, TAG_PROPERTIES.INNER_HTML], propsList),
```
Comment (doctyper) [inline src/Helmet.js]: Are you allowing `style={{ innerHTML: "" }}`?

Comment (cwelch5) [inline src/Helmet.js]: Ah, no, wasn't planning to.  Left this in by mistake.

File: src/Helmet.js (hunk 1 above)
Comment (cwelch5) [inline src/Helmet.js]: Hmmm yes, good find, curious why that still works.

**Comment by jamsea** [comment]
LGTM

**Pull request merged**